core/src/main/java/io/methvin/watcher/changeset/ChangeSetListener.java

```diff
@@ -44,6 +67,11 @@ public void onEvent(DirectoryChangeEvent event) {
         case OVERFLOW:
           throw new IllegalStateException("OVERFLOW not yet handled");
       }
+
+      if (timeout > 0 && currentTask != null) {
```

Comment:
I don't think we need `timeout > 0` since all we care about is there's a task to cancel.

Reply:
currentTask is null if we don't use onIdle